clib/dpd.js

```diff
@@ -2,9 +2,31 @@
 
   if (!window._dpd) window._dpd = {};
 
-  var root = window.location.protocol + '//' + window.location.hostname;
-  if (window.location.port !== '') {
-    root += ':' + window.location.port;
+  var root = dpdScriptElement().getAttribute('src')
+      .replace(/(.*)(\/.*\.js)$/, '$1')
+      .replace(/^[^\/]*.js$/, '');
+
+  function dpdScriptElement() {
+    // Only works in modern non-IE browsers
+    var element = document.currentScript;
+
+    if (!element) {
+      // Allows script to have a name other than dpd.js by assigning id="dpd.js" to script tag
+      element = document.getElementById('dpd.js');
```

Comment:
This is a bit strange to me. 
As most people won't setup like this id="dpd.js". 

Please remove document.getElementById('dpd.js') to save some processing time.